codalab/model/bundle_model.py

```diff
@@ -539,113 +543,114 @@ def make_condition(key, field, value):
                         target_datetime.timestamp()
                     )
-
-                clause = cl_bundle.c.uuid.in_(
-                    alias(
-                        select([cl_bundle_metadata.c.bundle_uuid]).where(
-                            and_(cl_bundle_metadata.c.metadata_key == 'created', subclause)
-                        )
-                    )
+                add_join(
+                    aliased_bundle_metadata,
+                    cl_bundle.c.uuid == aliased_bundle_metadata.c.bundle_uuid,
                 )
+                where_clause = and_(aliased_bundle_metadata.c.metadata_key == 'created', subclause)
             elif key == 'uuid_name':  # Search uuid and name by default
-                clause = []
-                clause.append(cl_bundle.c.uuid.like('%' + value + '%'))
-                clause.append(
-                    cl_bundle.c.uuid.in_(
-                        alias(
-                            select([cl_bundle_metadata.c.bundle_uuid]).where(
-                                and_(
-                                    cl_bundle_metadata.c.metadata_key == 'name',
-                                    cl_bundle_metadata.c.metadata_value.like('%' + value + '%'),
-                                )
-                            )
-                        )
+                aliased_bundle_metadata = aliased(cl_bundle_metadata)
+                add_join(
+                    aliased_bundle_metadata,
+                    cl_bundle.c.uuid == aliased_bundle_metadata.c.bundle_uuid,
+                )
+
+                where_clause = []
+                where_clause.append(cl_bundle.c.uuid.like('%' + value + '%'))
+                where_clause.append(
+                    and_(
+                        aliased_bundle_metadata.c.metadata_key == 'name',
+                        aliased_bundle_metadata.c.metadata_value.like('%' + value + '%'),
                     )
                 )
-                clause = or_(*clause)
+                where_clause = or_(*where_clause)
             elif key == '':  # Match any field
-                clause = []
-                clause.append(cl_bundle.c.uuid.like('%' + value + '%'))
-                clause.append(cl_bundle.c.command.like('%' + value + '%'))
-                clause.append(
-                    cl_bundle.c.uuid.in_(
-                        alias(
-                            select([cl_bundle_metadata.c.bundle_uuid]).where(
-                                cl_bundle_metadata.c.metadata_value.like('%' + value + '%')
-                            )
-                        )
-                    )
+                aliased_bundle_metadata = aliased(cl_bundle_metadata)
+                add_join(
+                    aliased_bundle_metadata,
+                    cl_bundle.c.uuid == aliased_bundle_metadata.c.bundle_uuid,
                 )
-                clause = or_(*clause)
+
+                where_clause = []
```

Comment:
seems like you can just write `or_(cl_bundle..., ..., ...)` directly instead of creating an array?

Reply:
Done.